### DISCUSSION TITLE
KeyError: 'eth_usd_price_feed'

### DISCUSSION THREAD

**Discussion by hiteshgupta2507** [smartcontractkit/full-blockchain-solidity-course-py#1616] [Q&A] [answered]
I am getting the following error when I run **brownie run scripts/deploy_lottery.py**
KeyError: 'eth_usd_price_feed'

LOCAL_BLOCKCHAIN_ENVIRONMENTS is set to ["developmemt"] but it still is looking for "eth_usd_price_feed" if I just run
brownie run scripts/deploy_lottery.py

github repo: [](https://github.com/hiteshgupta2507/SmartContract_Lottery)

**FYI:** `brownie run scripts/deploy_lottery.py --network rinkeby `works fine. The problem is with development.

Please advice.

**Comment by cromewar** [comment, marked as answer]
hello @hiteshgupta2507 
Please ve sure you don't have any typo on your helpful scripts and the network is setting up as `development` correctly, also remember that you have to add the else statement to deploy the mocks when you are on that network.
It would be easier to help if you share the code you are using here.

Cheers.

**Reply by hiteshgupta2507** [reply]
Hello @cromewar 

I have checked the helpful scripts and the network is set up as development correctly (as far as I can tell). My apologies, I thought that I had shared git repo link: https://github.com/hiteshgupta2507/SmartContract_Lottery

**Reply by cromewar** [reply]
OK I just checked out the code and the mocks is just for actual contract, VRF and LinkToken and so on, but you still need a `eth_usd_price_feed` for the development network, just duplicate the one for rinkeby and it should work right away.

**Reply by hiteshgupta2507** [reply]
It won't work coz if I copy eth_usd_price_feed from rinkeby into development the contract won't be found since at that address it exists on rinkeby chain. I still tried it though and got the error
`ContractNotFound: No contract deployed at 0x8A753747A1Fa494EC906cE90E9f37563A8AF630e`

But technically for development it shouldn't look for eth_usd_price_feed because since I am on development network it should just deploy mocks contract based on how get_contract function is defined. 

The function is defined exactly how @PatrickAlphaC has but still doesn't work for me in development network.
(Patrick's repo link: https://github.com/PatrickAlphaC/smartcontract-lottery/blob/main/scripts/helpful_scripts.py )

**Reply by hiteshgupta2507** [reply]
It's fixed. I found the error. Thanks for looking into the issue.
